src/plugins/graph/imports/components/axis/components/axis-or-legend-attribute-menu.tsx

```diff
@@ -77,57 +85,55 @@ const _AxisOrLegendAttributeMenu = ({ place, attributeId, target, portal, onOpen
+    <div className={`axis-legend-attribute-menu ${place}`} >
       <Menu boundary="scrollParent">
         {({ onClose, isOpen }) => {
           onOpenClose && onOpenClose(isOpen);
           onCloseRef.current = onClose;
           return (
-            <div ref={setDragNodeRef} style={overlayStyle} {...attributes} {...listeners}>
-              <MenuButton style={buttonStyle}>{attribute?.name}</MenuButton>
-              <MenuList>
-                { !data &&
-                  <MenuItem className="inactive">
-                    Link Data
-                  </MenuItem>
-                }
-                { data?.attributes?.map((attr, idx) => {
-                  //only show y attr that is not self, not the x axis, and not an already plotted Y
-                  const isCurrent = attr.id === attributeId;
-                  const isXAxis = (idx === 0);
-                  const isAPlottedYAttribute = yAttributesPlotted?.includes(attr.id);
-                  const showAttr = (!isCurrent && !isXAxis && !isAPlottedYAttribute);
-
-                  return (
-                    showAttr &&
-                    <MenuItem
-                      onClick={() => {onChangeAttribute(place, data, attr.id, attrId);}}
-                      key={attr.id}
-                    >
-                      {attr.name}
-                    </MenuItem>
-                  );
-                })}
-                { attribute &&
-                  <>
-                    <MenuDivider />
-                    { showRemoveOption &&
-                      <MenuItem onClick={() => onRemoveAttribute(place, attrId)}>
-                       {removeAttrItemLabel}
+            <Portal containerRef={parentRef}>
+              <div ref={setDragNodeRef} style={overlayStyle} {...attributes} {...listeners}>
+                <MenuButton style={buttonStyle}>{attribute?.name}</MenuButton>
+                <Portal containerRef={portalRef}>
+                  <MenuList ref={menuListRef}>
+                    { !data &&
+                      <MenuItem className="inactive">
+                        Link Data
                       </MenuItem>
                     }
-                    <MenuItem onClick={() => onTreatAttributeAs(place, attribute?.id, treatAs)}>
-                      {treatAs === "categorical" && t("DG.DataDisplayMenu.treatAsCategorical")}
-                      {treatAs === "numeric" && t("DG.DataDisplayMenu.treatAsNumeric")}
-                    </MenuItem>
-                  </>
-                }
-              </MenuList>
-            </div>
+                    { data?.attributes?.map((attr) => {
+                      const isCurrent = attr.id === attrId;
+                      const isPlottedX = dataConfig?.attributeID("x") === attr.id;
+                      const isPlottedY = yAttributesPlotted?.includes(attr.id);
+                      const showAttr = (!isCurrent && !isPlottedX && !isPlottedY);
```

Comment:
@kswenson and @dennisrcao - this represents a change I made to the filtering that was recently implemented. 

see the diff in [this commit](https://github.com/concord-consortium/collaborative-learning/pull/2024/commits/79157ed68c48d5676bab000b931498eb10f8430c). 

This addresses an issue that cropped up when I was reviewing.  If we had a table with columns x, y, and y2, and configured the graph to plot y2 on the x axis, we would lose the ability to change it back.  In my testing, this fixes that issue.